components/AppBar/src/MDCAppBar.m

```diff
@@ -308,22 +310,25 @@ - (void)viewDidLoad {
                             views:@{kBarStackKey : self.headerStackView}];
   [self.view addConstraints:horizontalConstraints];
 
+  CGFloat topMargin = kPreIOS11StatusBarHeight;
 #if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
   if (@available(iOS 11.0, *)) {
-    [self.headerStackView.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor
-                                                   constant:0].active = YES;
-  } else {
-#endif
-  [NSLayoutConstraint constraintWithItem:self.headerStackView
-                               attribute:NSLayoutAttributeTop
-                               relatedBy:NSLayoutRelationEqual
-                                  toItem:self.view
-                               attribute:NSLayoutAttributeTop
-                              multiplier:1
-                                constant:kPreIOS11StatusBarHeight].active = YES;
-#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
+    // We only get the actual status bar height if we're on iOS 11, because previously we were not
+    // adjusting the header height to make it smaller when the status bar is hidden.
+    // Note: We should switch to using self.view.safeAreaLayoutGuide.topAnchor once 11.1 is
+    // the min iOS 11 we support. 11.0 has a bug where the Safe Area is incorrectly calculated for
+    // this view.
+    topMargin = [UIApplication mdc_safeSharedApplication].statusBarFrame.size.height;
```

Comment:
Nit: I think it's better to use the maxY of the status Bar in case its frame does not start at 0?  The maxY is the offset into the window where the headerStackView should start.

```objectivec
CGRectGetMaxY([UIApplication mdc_safeSharedApplication].statusBarFrame)
```

Reply:
Done (even though I've never seen a status bar frame that doesn't start at 0 :shrug: